downstairs/src/region.rs

```diff
@@ -1053,6 +1014,136 @@ impl Region {
         Ok(())
     }
 
+    #[cfg(not(feature = "omicron-build"))]
+    #[allow(clippy::unused_async)]
```

Comment:
What uses async here? Or, what does not use async that we are allowing?

Reply:
At some point, we may have per-extent IO workers, so we deliberately left some functions as async.